python/tchannel/tchannel.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import sys
+import socket
+
+import tornado.ioloop
+import tornado.iostream
+
+from .tornado.connection import TornadoConnection
+from .exceptions import InvalidMessageException
+from .messages import CallRequestMessage
+
+
+log = logging.getLogger('tchannel')
+
+
+class TChannel(object):
+    """Manages inbound and outbound connections to various hosts."""
+
+    def __init__(self):
+        self.peers = {}
+        self.process_name = "%s[%s]" % (sys.argv[0], os.getpid())
```

Comment:
Make this an optional argument and generate this as the default value.

Also, it wouldn't be the worst idea to truncate `sys.argv[0]` to some length.